### ISSUE TITLE
Instances for Data.List.NonEmpty

### ISSUE THREAD

**Issue by phadej** [bos/aeson#362]
it's in base-4.9. We probably should depend on `semigroups` then.

**Comment by hvr** [comment]
How would you define the ToJSON/FromJSON instances? I assume you'd use the same serialisation as for `[a]` ? And you'd have to fail in the `FromJSON` on `[]`?

**Comment by phadej** [comment]
yeah, I'd probably fail on `[]` similarly as with `Natural` instance

**Comment by phadej** [comment]
@hvr, actually we already conditionally depend on `semigroups`

**Comment by bergmark** [comment]
SGTM
